src/emulator/remoteconfig/index.ts

```diff
@@ -0,0 +1,301 @@
+import * as utils from "../../utils";
+import { Constants } from "../constants";
+import { EmulatorInfo, EmulatorInstance, Emulators } from "../types";
+import { createApp } from "./server";
+import { EmulatorLogger } from "../emulatorLogger";
+import express = require("express");
+import * as fs from "fs";
+import { FirebaseError } from "../../error";
+import * as chokidar from "chokidar";
+import { cloneDeep } from "lodash";
+import { RemoteConfigCloudFunctions } from "./cloudFunctions";
+
+export interface RemoteConfigEmulatorArgs {
+  projectId: string;
+  port?: number;
+  host?: string;
+  template: Source | string;
+}
+
+export interface Source {
+  files: SourceFile[];
+}
+
+export interface SourceFile {
+  name: string;
+  content: string;
+}
+
+export interface ValidTemplate {
+  valid: boolean;
+  msg: string;
+}
+
+export class RemoteConfigEmulator implements EmulatorInstance {
+  private destroyServer?: () => Promise<void>;
+  private _app?: express.Express;
```

Comment:
Nit: This _Xyz style is a bit dated - its mostly used for pure JS implementations to denote something as private by convention (since for a long time, JS did not have any concept of private properties). Since typescript lets us actually declare fields private, theres no need to do this anymore.